### PR TITLE
coost: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **coost/3.0.1**

retry https://github.com/conan-io/conan-center-index/pull/12789.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/coost/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "coost")
+        self.cpp_info.set_property("cmake_target_name", "coost::co")
+        # TODO: back to global scope in conan v2 once legacy generators removed
```
Comment (czoido) [inline recipes/coost/all/conanfile.py:96]: Thanks a lot for the contribution! 
I think that being this a new recipe that is not adding test_v1 to test the legacy generators, we could get rid of the component.

Comment (toge) [reply recipes/coost/all/conanfile.py:96]: @czoido 
Thank you for reviewing it.
I forgot to delete it.

Comment (toge) [reply recipes/coost/all/conanfile.py]: fixed.

**Review by RubenRBS — APPROVED** [review]
I like how clea this recipe looks, great job @toge :)

**Pull request merged**